### PR TITLE
Make finding of NetCDF and HDF5 more explicit.

### PR DESCRIPTION
**Pull request by citibeth**
@skosukhin @adamjstewart 

This is a follow-on to #16492 and #10885.  I ran into the same problem as #16492 and ended up coming up with the same solution (because I didn't have the latest Spack).  My solution had some additional env var settings that help make things more explicit, and hopefully more bulletproof.

@adamjstewart wrote:
> We should probably fix HDF5 to install pkg-config files so that this patch won't be needed forever

Not sure I agree.  `pkg-config` is a mess of automagic that never worked very well and, more often than not, conflicts with Spack.  As long as we've been explicit about where HDF5 is to be found (via the `HDF5_DIR` env var), then I think disabling `pkg-config` is the more robust approach.

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
Just waiting for a review from @skosukhin

**Comment by skosukhin** [comment]
@adamjstewart LGTM

**Review by alalazo — APPROVED** [review]
Thanks @skosukhin

**Pull request merged**
